### PR TITLE
Skip bad font in test_font

### PR DESCRIPTION
**Pull request by hmaarrfk**
Fixes: https://github.com/vispy/vispy/issues/1771

Also uses fixtures so as not to change the global state of the the warnings filter.

### CODE REVIEW

**Comment by djhoese** [comment]
Thanks @hmaarrfk.

@kmuehlbauer @larsoner I was talking with a coworker who found out that vispy's tests technically don't need pytest. I recently added some tests and this PR adds tests that import `pytest` from the top of the module. This seems like the easiest time to ask if you are OK with me removing the fallback code in vispy's tests so that it will now require `pytest` to run.

**Comment by kmuehlbauer** [comment]
> This seems like the easiest time to ask if you are OK with me removing the fallback code in vispy's tests so that it will now require `pytest` to run.

@djhoese Removing code sounds good. I'm fine with whatever makes live easier .

**Comment by hmaarrfk** [comment]
yeah, you guys definitely require `nose` which is quite old, and will likely to stop being supported
https://nose.readthedocs.io/en/latest/#note-to-users

many projects including numpy and scipy have moved to completely supporting and requiring pytest
https://github.com/numpy/numpy/blob/master/test_requirements.txt#L2

**Comment by djhoese** [comment]
@hmaarrfk I'm 90% sure I've looked at this and we don't depend on `nose` but do depend on parts of numpy.testing that require nose.

**Comment by hmaarrfk** [comment]
Yeah, seems like you are right. I do see lots of logic in there that is trying to work around the myriad of different testing backends.

**Comment by larsoner** [comment]
Yes a requirement for pytest is fine

File: vispy/util/fonts/tests/test_font.py [hunk 1]
```diff
@@ -3,39 +3,42 @@
...
 
-def test_font_glyph():
+@pytest.mark.parametrize('face', ['OpenSans'] + sorted(sys_fonts))
+def test_font_glyph(face):
     """Test loading glyphs"""
-    # try both a vispy and system font
-    sys_fonts = set(list_fonts()) - set(_vispy_fonts)
-    # assert_true(len(sys_fonts) > 0) not always the case!
-    for face in ['OpenSans'] + sorted(list(sys_fonts)):
-        print(face)  # useful for debugging
```
Comment (hmaarrfk) [inline vispy/util/fonts/tests/test_font.py]: The print statement is no longer useful since the font name will appear sure to parameterization

Comment (djhoese) [reply vispy/util/fonts/tests/test_font.py]: Awesome!

File: vispy/util/fonts/tests/test_font.py [hunk 2]
```diff
@@ -3,39 +3,42 @@
...
-        glyphs_dict = dict()
-        chars = 'foobar^C&#'
-        for char in chars:
-            # Warning that Arial might not exist
-            with warnings.catch_warnings(record=True):
-                warnings.simplefilter('always')
-                _load_glyph(font_dict, char, glyphs_dict)
-        assert_equal(len(glyphs_dict), np.unique([c for c in chars]).size)
+    if face in known_bad_fonts:
+        pytest.skip()
```
Comment (hmaarrfk) [inline vispy/util/fonts/tests/test_font.py]:
```suggestion
        pytest.xfail()
```

**Pull request merged**